src/test/java/org/scijava/table/io/TableIOPluginTest.java

```diff
@@ -0,0 +1,122 @@
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package org.scijava.table.io;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.scijava.Context;
+import org.scijava.io.IOPlugin;
+import org.scijava.plugin.PluginInfo;
+import org.scijava.plugin.PluginService;
+import org.scijava.table.DefaultGenericTable;
+import org.scijava.table.Table;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class TableIOPluginTest {
+
+	private Context context;
+
+	@SuppressWarnings("rawtypes")
+	@Before
+	public void setUp() {
+		context = new Context();
+		PluginInfo<IOPlugin> info = PluginInfo.create(FakeTableIOPlugin.class,
+			IOPlugin.class);
+		context.service(PluginService.class).addPlugin(info);
+	}
+
+	@After
+	public void tearDown() {
+		context.dispose();
+		context = null;
+	}
+
+	@Test
+	public void testTableIOService() {
+		String tableFile = "fakeTableFile.csv";
+		TableIOService tableIOService = context.getService(TableIOService.class);
+		try {
+			Table<?, ?> data = tableIOService.open(tableFile);
+			assertTrue(Table.class.isAssignableFrom(data.getClass()));
+		}
+		catch (IOException exc) {
+			fail(exc.toString());
+		}
+	}
+
+	@Test
+	public void testTableIOServiceWithOptions() {
+		String tableFile = "fakeTableFile.csv";
+		TableIOService tableIOService = context.getService(TableIOService.class);
+		TableIOOptions options = new TableIOOptions()
+				.readColumnHeaders(true)
+				.readRowHeaders(true)
+				.columnDelimiter('-')
+				.columnType(0, Double.class);
+		try {
+			Table<?, ?> data = tableIOService.open(tableFile, options);
+			assertTrue(Table.class.isAssignableFrom(data.getClass()));
+		}
+		catch (IOException exc) {
+			fail(exc.toString());
+		}
+	}
+
+	@SuppressWarnings("rawtypes")
+	public static class FakeTableIOPlugin extends TableIOPlugin {
+
+		@Override
+		public boolean supportsOpen(String loc) {
+			return loc.endsWith("csv");
+		}
+
+		@Override
+		public boolean supportsSave(String loc) {
+			return loc.endsWith("csv");
+		}
+
+		@Override
+		public Table open(String loc) {
+			return new DefaultGenericTable();
+		}
```

Comment:
This method isn't necessary, as `TableIOPlugin` defines a default `open(String)` calling `open(String, TableIOOptions)`, no?